### ISSUE TITLE
Python multiprocessing not working with aer simulator on Ubuntu

### ISSUE THREAD

**Issue by Eatenbybug** [Qiskit/qiskit-terra#6966]
### Informations

- **Qiskit Aer version**: Qiskit Aer 0.8.2
- **Python version**: 3.7.11
- **Operating system**: Ubuntu 18.04.3 LTS

### What is the current behavior?
When I am trying to run VQE calculations in parallel using multiprocessing package in python, there is no output if I use aer_simulator as the backend. The program, however, runs fine if I use statevector_simulator.
In Windows 10, there is no problem with multiprocessing. 

### Steps to reproduce the problem
The following code illustrates the problem.
```python
import numpy as np

from qiskit import Aer
from qiskit.opflow import X, Z, I
from qiskit.utils import QuantumInstance, algorithm_globals
from qiskit.algorithms import VQE, NumPyMinimumEigensolver
from qiskit.algorithms.optimizers import COBYLA, L_BFGS_B, SLSQP
from qiskit.circuit.library import TwoLocal

from qiskit.providers.aer import QasmSimulator
from qiskit.providers.aer.noise import NoiseModel
from qiskit.test.mock import FakeVigo

import time

import multiprocessing as mp


def H2_solver():
    #Construct the Hamiltonian for H2
    H2_op = (-1.052373245772859 * I ^ I) + \
        (0.39793742484318045 * I ^ Z) + \
        (-0.39793742484318045 * Z ^ I) + \
        (-0.01128010425623538 * Z ^ Z) + \
        (0.18093119978423156 * X ^ X)
        
    optimizer = SLSQP(maxiter=20)
    ansatz = TwoLocal(rotation_blocks='ry', entanglement_blocks='cz')


    backend = Aer.get_backend('statevector_simulator')
    
    qi = QuantumInstance(backend=backend)
    
    vqe = VQE(ansatz, optimizer, quantum_instance=qi)
    
    
    result = vqe.compute_minimum_eigenvalue(operator=H2_op)
    return result.eigenvalue

results = []
def get_result(result):
    #helper function to write returned result into results list.
    global results
    results.append(result)

if __name__ == '__main__':
    # Step 1: Init multiprocessing.Pool()
    pool = mp.Pool(2)
    

    # Step 2: `pool.apply` the `howmany_within_range()`
    for i in range(2):
        pool.apply_async(H2_solver, callback = get_result)
    pool.close() 
    pool.join()
    print(results)
    

```
The returned result is just empty list "[ ]"
 

### What is the expected behavior?

Expected result should be list of expectation values like [(-1.8572750286519004+0j), (-1.8572750233347164+0j)]. If the backend in the code is changed to 
`backend = Aer.get_backend('statevector_simulator')`,
one can see the correct output.

### Suggested solutions
I tested the same code in Windows 10 and no issue there.

**Comment by kdk** [comment]
Qiskit in general will already try to take advantage of multiprocessing where it can, and I believe this can lead to issues when attempting to add another layer of multiprocesing on top (e.g. the issues described in #3426 ). A possible solution might be to disable Qiskit's internal parallelization (setting `os.environ['QISKIT_IN_PARALLEL'] = 'TRUE'` ) in your application.

**Comment by jakelishman** [comment]
Presumably there is some exception being thrown by Aer, which is why the callback function is not getting run in these situations.  To dig down and work out what the full problem is, we'd need to see the exception being thrown.  You can check if the result of an `apply_async` call was successful by doing `async_result = pool.apply_async(...)`, and then `async_result.successful()` (after joining).  If it was unsuccessful, then doing `async_result.get()` should re-raise the exception that caused the failure.

**Comment by woodsp-ibm** [comment]
Algorithms now use primitive and the older VQE here, that was based on QuantumInstance is deprecated and will soon be removed. I am going to close this as no longer relevant,
